lib/rules/space-in-parens.js

```diff
@@ -184,9 +188,9 @@ module.exports = function(context) {
 
     return {
         "Program": function checkParenSpaces(node) {
-            var tokens = node.tokens;
-            var prevToken, nextToken;
+            var tokens, prevToken, nextToken;
             exceptions = getExceptions();
+            tokens = sourceCode.tokensAndComments;
```

Comment:
Yes. However, as you mentioned in the issue, this change likely needs to be made in a bunch of rules, and in that case we want the array cached and directly accessible without a function call (which would be additional overhead on top of the property access. 

So we pay the cost up front, store the result, and basically get it for free every other time is used vs. calling a function that always had a cost.